wrappers/python/openmm/app/internal/xtc_utils/src/xtc.cpp

```diff
@@ -44,7 +45,7 @@ static size_t Zf(size_t yidx, size_t nframes) {
 // Helper struct to manage the XDRFILE pointer
 struct XDRFILE_RAII {
     XDRFILE* xd;
-    XDRFILE_RAII(const char* filename, const char* mode) : xd(xdrfile_open(filename, mode)) {
+    XDRFILE_RAII(std::string filename, std::string mode) : xd(xdrfile_open(filename.data(), mode.data())) {
```

Comment:
To convert a C++ string to a `char*`, call `.c_str()`.  The pointer returned by `.data()` isn't guaranteed to be null terminated.

Reply:
The evil person that wrote the function read_xtc_natoms made it so that it takes char*, refusing to accept filename.c_str().
It allows me to do this, though:
```c++
read_xtc_natoms(const_cast<char*>(filename.c_str()), &natoms)
```